hack/tests/test_default_deploy.sh

```diff
@@ -0,0 +1,578 @@
+#!/bin/bash
+TEST_SOURCE=${BASH_SOURCE}
+TEST_DIR=$(dirname "${BASH_SOURCE}")
+
+function tests.setup {
+  #initial setup required for all test scenarios
+  oc create -f $SOURCE_ROOT/metrics-deployer-setup.yaml &> /dev/null
+  oadm policy add-role-to-user edit system:serviceaccount:${TEST_PROJECT}:metrics-deployer
+  oadm policy add-cluster-role-to-user cluster-reader system:serviceaccount:${TEST_PROJECT}:heapster
+}
+
+function tests.teardown {
+  #clean up required after the tests have run.
+  oadm policy remove-cluster-role-from-user cluster-reader system:serviceaccount:${TEST_PROJECT}:heapster
+}
+
+function checkDeployer {
+  Info "Checking the deployer."
+
+  DEPLOYER_START=$(date +%s)
+
+  deployer_pod=`oc get pods | grep -i metrics-deployer`
+
+  pods=`oc get pods | grep -i metrics-deployer` || true
+
+  while read -r pod; do
+   name=`echo $pod | awk '{print $1}'`
+   state=`echo $pod | awk '{print $3}'`
+   if [[ $state == "Pending" ]]; then
+     podName=$name
+     break
+   fi
+  done <<< "$pods"
+
+  if [[ -z ${podName-} ]]; then
+    Fail "No deployer pod currently being deployed"
+  fi
+
+  while : 
+  do
+    if [[ $(($(date +%s) - $DEPLOYER_START)) -ge 120 ]]; then
+      Fail "Deployer Pod took longer than the timeout of 120 seconds"
+    fi
+
+    deployer_pod=`oc get pods | grep -i metrics-deployer`
+    deployer_status=`echo $deployer_pod | awk '{print $3}'`
+
+    Debug "The current status of the deployer:$deployer_status"
+
+
+    if [ -z "$deployer_pod" ] || [[ $deployer_status == "Failed" ]] || [[ $deployer_status == "Error" ]]; then
+      Fail "Deployer Pod was not deployed. `echo $deployer_pod | awk '{print $3}'`"
+    fi
+
+    if [[ $deployer_status == "Terminating" ]] || [[ $deployer_status == "Completed" ]]; then
+      Info "The deployer was deployed in $(($(date +%s) - $DEPLOYER_START)) seconds."
+      return
+    fi
+
+    sleep 1
+  done
+}
+
+# $1 the deployments name
+# $2 the number of replicas to check
+function checkDeployment {
+  name=$1
+  replicas=$2
+
+  echo
+  Info $SEPARATOR
+  Info "Checking the deployment of $name. Please wait."
+  CHECK_START=$(date +%s)
+
+  while : 
+  do
+    if [[ $(($(date +%s) - $CHECK_START)) -ge 120 ]]; then
+      Fail "$name took longer than the timeout of 120 seconds"
+    fi
+
+    states=`oc get pods | grep -i $name | awk '{print $3}'` || true
```

Comment:
Not sure how far you want to take this, but 'oc' does except golang templates to some capacity which would preclude you from having to pipe this through grep and awk